### PR TITLE
Add a loading screen when starting a mission before the MissionInProgress Activity

### PR DESCRIPTION
**Pull request by GabrielFleischer**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivityTest.kt [hunk 1]
```diff
@@ -1,36 +1,68 @@
-/*
- * Copyright (C) 2021  Drone3D-Team
- * The license can be found in LICENSE at root of the repository
- */
-
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivityTest.kt:5]: I think we agreed the display the header in every file.

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import ch.epfl.sdp.drone3d.ui.ToastHandler
+import ch.epfl.sdp.drone3d.ui.map.MissionInProgressActivity
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.AndroidEntryPoint
+import io.reactivex.disposables.Disposable
+import timber.log.Timber
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MissionStartActivity : AppCompatActivity() {
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:23]: A comment explaining the purpose of the activity would be appreciated

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_mission_start)
+
+        val progressBar = findViewById<ProgressBar>(R.id.mission_start_progress).apply {
+            max = totalProgress()
+        }
+
+        val progressText = findViewById<TextView>(R.id.mission_start_text)
+
+        droneService.getData().getDroneStatus().observe(this) { status ->
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:50]: Correct me if I'm wrong but, I belive that when you provide the activity to a observer, it stops observing when the class is destroyed automatically.

**Review by jonasblanc — CHANGES_REQUESTED** [review]

**Review by Anghul — APPROVED** [review]
I have nothing to add to Jonas' comment. Everything else seems fine to me

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivityTest.kt (hunk 1 above)
Comment (GabrielFleischer) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivityTest.kt:5]: It was removed for some reason

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt (hunk 3 above)
Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:50]: I will onvestigate that

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                else -> R.string.mission_state_unknown
+            }
+        )
+
+    private fun setupMission(missionPath: List<LatLng>?) {
+        if(missionPath == null) {
+            ToastHandler.showToastAsync(this, R.string.mission_null)
+            finish()
+        } else {
+            val droneMission = DroneUtils.makeDroneMission(missionPath, 20f)
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:91]: Altitude should be handled later, putting a constant there could have made things easier

**Review by Squalene — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt (hunk 3 above)
Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:50]: That's true

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt (hunk 4 above)
Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt:98]: Done

**Review by jonasblanc — APPROVED** [review]

**Pull request merged**